src/code/z_scene.c

```diff
@@ -3,116 +3,135 @@
 
 RomFile sNaviMsgFiles[];
 
-s32 Object_Spawn(ObjectContext* objectCtx, s16 objectId) {
+/**
+ * Spawn an object file of a specified ID that will persist through room changes.
+ *
+ * This only starts loading the object, the data will not be available immediately when the function returns.
+ *
+ * Used for gameplay_keep, Link's object, and the subkeep if any.
+ *
+ * @return The new object slot corresponding to the requested object ID.
+ *
+ * @note This function is not meant to be called externally to spawn object files on the fly.
+ * When an object is spawned with this function, all objects that come before it in the entry list will be treated as
+ * persistent, which will likely cause the memory available for object space to run out.
+ * This function is only meant to be called internally on scene load, before the object list from any room is processed.
```

Comment:
Good point about the slots, I'll amend the comment

I don't think capital letters are good practice, if it's so important we should more properly use `@warning`
(doxygen has `@remark`, `@note`, `@attention`, `@warning`)

<details>

```c
/**
 * Spawn an object file of a specified ID that will persist through room changes.
 *
 * This only starts loading the object, the data will not be available immediately when the function returns.
 *
 * Used for gameplay_keep, Link's object, and the subkeep if any.
 *
 * @return The new object slot corresponding to the requested object ID.
 *
 * @remark This function is not meant to be called externally to spawn object files on the fly.
 * @note When an object is spawned with this function, all objects that come before it in the entry list will be treated as
 * @attention persistent, which will likely cause the memory available for object space to run out.
 * @warning This function is only meant to be called internally on scene load, before the object list from any room is processed.
 */
```

![image](https://user-images.githubusercontent.com/5306619/172933247-2dc9af58-3edb-40ef-a257-e9cc60a7dd22.png)

</details>

Reply:
I think theyre fine to use when emphasis is needed. I dont want to start using yet another doxygen tag for this, it just makes it read nicer with the emphasis that is intended in the message lol

Reply:
I really don't like documentation screaming at me or showing emotion

This function has been wrongly used for a while, but just naming it with `Persistent` and the `@note` is enough imo to document the behavior.
And when I'll document the rest of object handling I'll add an example (or just a `@see`) of how to actually request objects be loaded I guess

Reply:
Guess others will have to weigh in, cause I just dont agree. When we dont have access to things like italics and bold letters, caps are fine to use for important words like NOT, imo

Reply:
Oh we can do italics and bold
```c
@note *This function* is **not** meant to
```
![image](https://user-images.githubusercontent.com/5306619/172936620-1b65cf6e-c3ef-43e9-9992-f000e5137d12.png)

Not that I think it's appropriate here either but I guess it's still more professional looking than caps

Reply:
I mean within the code itself, we cant italicize or bold. I personally dont care too much for how doxygen output looks, the appearance in the code is a priority for me. And the caps gives the exact emphasis that I want on the comments

Some of our existing docs already do this, and even gbi from nintendo does. I dont see it as "unprofessional", just making use of what you have to get the correct tone across

Reply:
And for me the screaming tone is incorrect, as documentation shouldn't carry emotion

More opinions needed :smile:

Reply:
As I said in #1265 , I think it's worth having something stronger than `@note` for this. I also don't see why we shouldn't use Markdown, at least, since it's used on Discord, GitHub, in the README.md, ...